Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -671,7 +671,7 @@ void PcapLiveDevice::setDeviceMtu()
 	struct ifreq ifr;
 
 	memset(&ifr, 0, sizeof(ifr));
-	strncpy(ifr.ifr_name, m_Name, sizeof(ifr.ifr_name));
+	strncpy(ifr.ifr_name, m_Name, sizeof(ifr.ifr_name) - 1);

```

Comment:
why is it required to reduce the size by 1? My understanding is that `sizeof(ifr.ifr_name)` should give the size of the name

Reply:
`Sizeof` returns the max size of buffer including trailing '\0'. In case the length of `m_Name` is more than or equal to `sizeof(ifr.ifr_name)` then `strncpy` will fill all bytes of the buffer but the last byte will not be '\0' so the destination string is not null-terminated. Look into man on strncpy. Reducing the size by 1 we guarantee the destination string will be null-terminated.

Reply:
ok got it, thanks for the explanation. I'll merge the PR